File: tl/domain/post.go
package domain

import "time"

type Post struct {
	ID int `gorm:"primary_key"`
	V  int `gorm:"not null;type:bigint"`

	ParentId         int       `gorm:"not null"`
	Title            string    `gorm:"not null;size:150"`
	Post             string    `gorm:"not null;size:256000"`
	CreateTime       time.Time `gorm:"not null;type:datetime"`
	LastModifiedTime time.Time `gorm:"not null;type:datetime"`
	Status           int       `gorm:"not null"`
	NoTags           int       `gorm:"not null"`
}

File: src/main/kotlin/com/ariane/thirdlab/domains/Post.kt
package com.ariane.thirdlab.domains

import java.io.Serializable;
import java.util.*
import javax.persistence.*

@Entity
class Post() : Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    var id: Long = 0L

    @Version
    @Column(nullable = false, name = "v")
    var v: Long = 0

    @Column(nullable = false)
    var parentId: Long = 0

    @Column(nullable = false, length = 150)
    var title: String = ""

    @Column(nullable = false, length = 256000)
    var post: String = ""

    @Column(nullable = false)
    var createTime: Date = Date(0)

    @Column(nullable = false)
    var lastModifiedTime: Date = Date(0)

    @Column(nullable = false)
    var status: Int = 0

    @Column(nullable = false)
    var noTags: Int = 0

    @Column(nullable = false)
    var tagId: Long = 0
}